src/main/java/com/nha/abdm/wrapper/common/dataPackaging/encryption/EncryptionController.java

```diff
@@ -0,0 +1,169 @@
+/* (C) 2024 */
+package com.nha.abdm.wrapper.common.dataPackaging.encryption;
+
+import static com.nha.abdm.wrapper.common.dataPackaging.Constants.*;
+
+import com.nha.abdm.wrapper.common.dataPackaging.keys.KeyController;
+import com.nha.abdm.wrapper.common.dataPackaging.keys.KeyMaterial;
+import com.nha.abdm.wrapper.hip.hrp.dataTransfer.requests.HIPRequestBundle;
+import com.nha.abdm.wrapper.hip.hrp.dataTransfer.requests.callback.BundleResponseHIP;
+import com.nha.abdm.wrapper.hip.hrp.dataTransfer.requests.callback.HIPHealthInformationRequest;
+import java.math.BigInteger;
+import java.security.KeyFactory;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Security;
+import java.util.Arrays;
+import javax.crypto.KeyAgreement;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bouncycastle.asn1.x9.X9ECParameters;
+import org.bouncycastle.crypto.digests.SHA256Digest;
+import org.bouncycastle.crypto.ec.CustomNamedCurves;
+import org.bouncycastle.crypto.engines.AESEngine;
+import org.bouncycastle.crypto.generators.HKDFBytesGenerator;
+import org.bouncycastle.crypto.modes.GCMBlockCipher;
+import org.bouncycastle.crypto.params.AEADParameters;
+import org.bouncycastle.crypto.params.HKDFParameters;
+import org.bouncycastle.crypto.params.KeyParameter;
+import org.bouncycastle.jce.interfaces.ECPublicKey;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.jce.spec.ECParameterSpec;
+import org.bouncycastle.jce.spec.ECPrivateKeySpec;
+import org.bouncycastle.jce.spec.ECPublicKeySpec;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestBody;
+
+public class EncryptionController {
+  private static final Logger log = LogManager.getLogger(EncryptionController.class);
+  @Autowired KeyController keyController;
+
+  public EncryptionResponse encrypt(
+          @RequestBody HIPHealthInformationRequest hipHealthInformationRequest, BundleResponseHIP bundleResponse)
```

Comment:
Why `RequestBody` is needed here?

Reply:
For encryption, i need the bundle response from the facility and also the receiver public keys from HIPHealthInformationRequest to encrypt the FHIR bundle

Reply:
Oh, I get that, it is not needed here.